pkg/operator/sync.go

```diff
@@ -244,6 +246,96 @@ func (optr *Operator) syncRenderConfig(_ *renderConfig) error {
 		return fmt.Errorf("refusing to read images.json version %q, operator version %q", imgs.ReleaseVersion, optrVersion)
 	}
 
+	// handle image registry certificates.
+	// parse these, add them to ctrlcfgspec and then handle these in the daemon write to disk function
+	cfg, err := optr.configClient.ConfigV1().Images().Get(context.TODO(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	imgRegistryUsrData := []v1.ImageRegistryBundle{}
+	if cfg.Spec.AdditionalTrustedCA.Name != "" {
+		klog.Infof("got image.config.openshift.io/cluster, additionalCA: %s", cfg.Spec.AdditionalTrustedCA.Name)
+		cm, err := optr.clusterCmLister.ConfigMaps("openshift-config").Get(cfg.Spec.AdditionalTrustedCA.Name)
+		if err != nil {
+			klog.Warningf("could not find configmap specified in image.config.openshift.io/cluster with the name %s", cfg.Spec.AdditionalTrustedCA.Name)
+		}
+		for key := range cm.Data { // default to binay data if it is there, else use data
+			if bd, bdok := cm.BinaryData[key]; bdok {
+				imgRegistryUsrData = append(imgRegistryUsrData, v1.ImageRegistryBundle{
+					File: key,
+					Data: bd,
+				})
+			} else if d, dok := cm.Data[key]; dok {
+				raw, err := base64.StdEncoding.DecodeString(d)
+				if err != nil {
+					imgRegistryUsrData = append(imgRegistryUsrData, v1.ImageRegistryBundle{
+						File: key,
+						Data: []byte(d),
+					})
+				} else {
+					imgRegistryUsrData = append(imgRegistryUsrData, v1.ImageRegistryBundle{
+						File: key,
+						Data: raw,
+					})
+				}
+			}
+		}
+	}
+
+	imgRegistryData := []v1.ImageRegistryBundle{}
+	cm, err := optr.clusterCmLister.ConfigMaps("openshift-config-managed").Get("image-registry-ca")
+	if err != nil {
+		klog.Warningf("could not find configmap image-registry-ca, cluster-image-registry-operator might not be installed.")
```

Comment:
Do we really want to log a warning for this? If the image registry is not installed with the cluster, the user has nothing to worry about, so I think warning them could be misleading. Maybe an info or debug log plus a slight message rephrase?

Reply:
I will change it ti a debug, I think having some sort of message is helpful for changed behavior.